### ISSUE TITLE
Feature Enhancement: Be able to upload multiple streams

### ISSUE THREAD

**Issue by arman-g** [robinrodricks/FluentFTP#1228]
Currently the library supports uploading multiple files using a collection of either the file paths or the FileInfo objects. 

It would be really good to be able to upload multiple streams. This is useful when the files come from an external source (e.g. Azure blob storage etc.).

Regards,
AG

**Comment by FanDjango** [comment]
Could you not wrap this yourself as a multiple incarnation of a FluentFTP single stream API call?

**Comment by FanDjango** [comment]
> but I am not sure how single stream API call will handle file names.

Yes, I suppose you would be duplicating some stuff already in FluentFTP.

> Or just send each stream in a loop but this is not efficient.

Wait a moment. 

> supports uploading multiple files using a collection

internally, these are transferred **not** simultaneously. Rather: One after the other, right? So what's the difference looping them yourself?

**Comment by arman-g** [comment]
I see, if internally each file is transferred individually than I can just do it in a loop. I had the impression that it is one transfer and in case of failure the FTP server is responsible for the cleanup.

**Comment by FanDjango** [comment]
I have moved this to the wiki: https://github.com/robinrodricks/FluentFTP/wiki/Future-Plans
